src/datastore/postgres-store.ts

```diff
@@ -463,6 +481,61 @@ export class PgDataStore extends (EventEmitter as { new (): DataStoreEventEmitte
     this.emitAddressTxUpdates(data);
   }
 
+  getUnresolvedSubdomain(tx_id: string): Promise<FoundOrNot<DbBNSSubdomain>> {
+    return this.query(async client => {
+      const queryResult = await this.pool.query(
+        `
+        SELECT *
+        FROM subdomains
+        WHERE tx_id = $1
```

Comment:
This probably needs a `canonical = true` condition otherwise inconsistencies may happen during chain re-orgs.